src/libxtp/gwbse/sigma_cda.cc

```diff
@@ -0,0 +1,102 @@
+/*
+ *            Copyright 2009-2020 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <votca/tools/constants.h>
+#include <votca/xtp/gw.h>
+#include <votca/xtp/sigma_cda.h>
+
+namespace votca {
+namespace xtp {
+
+void Sigma_CDA::PrepareScreening() {
+  GaussianQuadrature::options opt;
+  opt.homo = _opt.homo;
+  opt.order = _opt.order;
+  opt.qptotal = _qptotal;
+  opt.qpmin = _opt.qpmin;
+  opt.rpamin = _opt.rpamin;
+  opt.quadrature_scheme = _opt.quadrature_scheme;
+  _gq.configure(opt, _rpa);
+}
+
+// This function is used in the calculation of the residues
+// This calculates eps^-1 (inverse of the dielectric function) for complex
+// frequencies of the kind omega = delta + i*eta
+double Sigma_CDA::CalcDiagContribution(const Eigen::RowVectorXd& Imx_row,
```

Comment:
I think this `.row()` is technically not a `RowVectorXd&`

Reply:
What is it then? `MatrixXd&`?

Reply:
`const Ref<const MatrixXd>&`

Reply:
changed